src/main/java/net/datafaker/annotations/FakeResolver.java

```diff
@@ -9,26 +9,33 @@
 
 public class FakeResolver<T> {
 
+    private static final JavaObjectTransformer jTransformer = new JavaObjectTransformer();
+
     private final Class<T> clazz;
 
     public FakeResolver(Class<T> clazz) {
         this.clazz = clazz;
     }
 
-    public T generate(Schema<Object, ?> schema) {
+    public T generateFromDefaultSchema() {
         checkFakeAnnotation(clazz);
 
         FakeForSchema fakeForSchemaAnnotation = clazz.getAnnotation(FakeForSchema.class);
-        Schema<Object, ?> useSchema = getSchema(schema, fakeForSchemaAnnotation.value());
+        Schema<Object, ?> useSchema = getSchema(fakeForSchemaAnnotation.value());
 
-        JavaObjectTransformer jTransformer = new JavaObjectTransformer();
         return (T) jTransformer.apply(clazz, useSchema);
     }
 
-    private Schema<Object, ?> getSchema(Schema<Object, ?> schema, String[] pathToSchema) {
-        if (schema != null) {
-            return schema;
-        } else if (pathToSchema.length != 0) {
+    public T generateFromCustomSchema(Schema<Object, ?> schema) {
+        if (Objects.isNull(schema)) {
+            return generateFromDefaultSchema();
+        }
+
+        return (T) jTransformer.apply(clazz, schema);
+    }
+
+    private Schema<Object, T> getSchema(String[] pathToSchema) {
```

Comment:
why do we need an array here if we use only the first element?
I guess just `String` should be enough

Reply:
Changed value type from array to a single value.